torch/_refs/__init__.py

```diff
@@ -3275,6 +3275,20 @@ def new_empty_strided(
     )
 
 
+@register_decomposition(torch.ops.aten.zeros)
+def zeros(
+    size: ShapeType,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    layout: Optional[torch.layout] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+) -> TensorLikeType:
+    return torch.full(
+        size, 0, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory
```

Comment:
How does this work? 

```
torch.zeros((3, 3)).dtype
: torch.float32

torch.full((3, 3), 0).dtype
: torch.int64
```

Does the dtype passed to `full` need to be the default floating point dtype instead of `None`?

Reply:
Also, it's an interesting quirk that we permit integer->bool Python type casts for numbers

Reply:
Not only we allow `integer->bool` Python casts, but people have used them quite a bit throughout refs. This is not my cup of tea, but I guess it makes the code slightly more compact.

Reply:
See updated code...

zeros' ref is using fill_value of False, ones' ref is  fill_value of True. 
Values of False and True can be casted to float or int, by just setting the expected dtype in torch.full.